# Importando librerias

In [2]:
import pandas as pd
import numpy as np

### Warnings

In [3]:
import warnings
warnings.filterwarnings('ignore')

# Carga de los datos

In [4]:
prov_accesos_cada_100_hogares_fijo=pd.read_csv('./pagina/Penetración de Internet fijo (accesos por cada 100 hogares).csv')

na_accesos_cada_100_hogares_fijo=pd.read_csv('./pagina/Penetración por hogares nacional de Internet fijo.csv')

na_accesos_b_ancha_angosta_fijo= pd.read_excel('./pagina/Internet_BAF.xlsx', sheet_name='Totales')

prov_accesos_b_ancha_angosta_fijo = pd.read_excel('./pagina/Internet_BAF.xlsx', sheet_name='Dial-BAf')

na_accesos_tipo_tecnologia_fijo= pd.read_excel('./pagina/Internet_Accesos-por-tecnologia.xlsx', sheet_name='Totales')

prov_accesos_tipo_tecnologia_fijo = pd.read_excel('./pagina/Internet_Accesos-por-tecnologia.xlsx', sheet_name='Accesos Por Tecnología')

na_velocidad_media_bajada= pd.read_excel('./pagina/historico_velocidad_internet.xlsx', sheet_name='Totales')

prov_velocidad_media_bajada = pd.read_excel('./pagina/historico_velocidad_internet.xlsx', sheet_name='Velocidad % por prov')

na_dis_accesos_rango_vel_bajada= pd.read_excel('./pagina/Internet_Accesos-por-velocidad.xlsx', sheet_name='Totales')

prov_dis_accesos_rango_vel_bajada = pd.read_excel('./pagina/Internet_Accesos-por-velocidad.xlsx', sheet_name='Accesos por velocidad')

prov_dis_accesos_todo_vel_bajada = pd.read_excel('./pagina/Internet_Accesos-por-velocidad_provincia_sin rangos.xlsx', sheet_name='Velocidad_sin_Rangos')

na_ingresos_operadores_fijo= pd.read_excel('./pagina/Internet_Ingresos.xlsx', sheet_name='Ingresos')

loc_dis_accesos_todo_vel_bajada_sin_t = pd.read_excel('./pagina/Internet_Accesos-por-velocidad_por loc.xlsx', sheet_name='internet_Accesos_por_loc')

loc_accesos_tipo_tecnologia_fijo_sin_t = pd.read_excel('./pagina/Internet-Accesos-por-tecno_por_loc.xlsx', sheet_name='Accesos_tecnologia_localidad')

loc_listado_conectividad_conexion_sin_t = pd.read_excel('./pagina/mapa_conectividad.xlsx', sheet_name='Hoja3')

# Limpieza y corrección de tipos de dato

### Penetración de Internet fijo (accesos por cada 100 hogares)
Número de accesos al servicio de Internet fijo por cada 100 hogares por provincia


In [5]:
prov_accesos_cada_100_hogares_fijo.head()

Año  Trimestre        Provincia Accesos por cada 100 hogares  Unnamed: 4  \
0  2022          4     Buenos Aires                        78,94         NaN   
1  2022          4  Capital Federal                       122,73         NaN   
2  2022          4        Catamarca                        65,77         NaN   
3  2022          4            Chaco                        43,62         NaN   
4  2022          4           Chubut                        85,57         NaN   

   Unnamed: 5  Unnamed: 6  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN

Vemos que el df tiene 3 columnas vacias y que 'Accesos por cada 100 hogares' no tiene el formato correcto

In [6]:
prov_accesos_cada_100_hogares_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 7 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           864 non-null    int64  
 1   Trimestre                     864 non-null    int64  
 2   Provincia                     864 non-null    object 
 3   Accesos por cada 100 hogares  864 non-null    object 
 4   Unnamed: 4                    0 non-null      float64
 5   Unnamed: 5                    0 non-null      float64
 6   Unnamed: 6                    0 non-null      float64
dtypes: float64(3), int64(2), object(2)
memory usage: 47.4+ KB


Elimino esas columanas vacias y cambiamos el formato

In [7]:
prov_accesos_cada_100_hogares_fijo=prov_accesos_cada_100_hogares_fijo.drop(['Unnamed: 4','Unnamed: 5','Unnamed: 6'],axis=1)

prov_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'] = prov_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'].str.replace(',', '.', regex=True)
prov_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'] = pd.to_numeric(prov_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'], errors='coerce')

In [8]:
prov_accesos_cada_100_hogares_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           864 non-null    int64  
 1   Trimestre                     864 non-null    int64  
 2   Provincia                     864 non-null    object 
 3   Accesos por cada 100 hogares  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


In [9]:
prov_accesos_cada_100_hogares_fijo.head()


Año  Trimestre        Provincia  Accesos por cada 100 hogares
0  2022          4     Buenos Aires                         78.94
1  2022          4  Capital Federal                        122.73
2  2022          4        Catamarca                         65.77
3  2022          4            Chaco                         43.62
4  2022          4           Chubut                         85.57

In [10]:
prov_accesos_cada_100_hogares_fijo.to_csv("./csv_corregidos/prov_accesos_cada_100_hogares_fijo.csv", index=False)

### Penetración por hogares nacional de Internet fijo
Serie trimestral de la penetración del Internet fijo en la métrica por cada 100 hogares


In [11]:
na_accesos_cada_100_hogares_fijo.head()


Año  Trimestre Accesos por cada 100 hogares Accesos por cada 100 hab  \
0  2022          4                        77,21                    24,15   
1  2022          3                        76,64                    23,95   
2  2022          2                        75,97                    23,72   
3  2022          1                        73,88                    23,05   
4  2021          4                        73,18                    22,81   

         Periodo  
0   Oct-Dic 2022  
1  Jul-Sept 2022  
2   Abr-Jun 2022  
3   Ene-Mar 2022  
4   Oct-Dic 2021

vemos que 'Accesos por cada 100 hogares' y 'Accesos por cada 100 hab' no tienen el tipo de dato correcto

In [12]:
na_accesos_cada_100_hogares_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           36 non-null     int64 
 1   Trimestre                     36 non-null     int64 
 2   Accesos por cada 100 hogares  36 non-null     object
 3   Accesos por cada 100 hab      36 non-null     object
 4   Periodo                       36 non-null     object
dtypes: int64(2), object(3)
memory usage: 1.5+ KB


In [13]:
na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'] = na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'].str.replace(',', '.', regex=True)

na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'] = pd.to_numeric(na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hogares'], errors='coerce')

na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hab'] = na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hab'].str.replace(',', '.', regex=True)

na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hab'] = pd.to_numeric(na_accesos_cada_100_hogares_fijo['Accesos por cada 100 hab'], errors='coerce')

In [14]:
na_accesos_cada_100_hogares_fijo.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           36 non-null     int64  
 1   Trimestre                     36 non-null     int64  
 2   Accesos por cada 100 hogares  36 non-null     float64
 3   Accesos por cada 100 hab      36 non-null     float64
 4   Periodo                       36 non-null     object 
dtypes: float64(2), int64(2), object(1)
memory usage: 1.5+ KB


In [15]:
na_accesos_cada_100_hogares_fijo.head()


Año  Trimestre  Accesos por cada 100 hogares  Accesos por cada 100 hab  \
0  2022          4                         77.21                     24.15   
1  2022          3                         76.64                     23.95   
2  2022          2                         75.97                     23.72   
3  2022          1                         73.88                     23.05   
4  2021          4                         73.18                     22.81   

         Periodo  
0   Oct-Dic 2022  
1  Jul-Sept 2022  
2   Abr-Jun 2022  
3   Ene-Mar 2022  
4   Oct-Dic 2021

In [16]:
na_accesos_cada_100_hogares_fijo.to_csv("./csv_corregidos/na_accesos_cada_100_hogares_fijo.csv", index=False)

### Total nacional de accesos a Internet fijo por banda ancha y banda angosta
Número total de accesos al servicio de Internet fijo por banda ancha y banda angosta (trimestral)


In [17]:
na_accesos_b_ancha_angosta_fijo.head()

Año  Trimestre  Banda ancha fija  Dial up     Total        Periodo
0  2022          4          11195668    12446  11208114   Oct-Dic 2022
1  2022          3          11078691    12437  11091128  Jul-Sept 2022
2  2022          2          10946248    12436  10958684   Abr-Jun 2022
3  2022          1          10611390    12619  10624009   Ene-Mar 2022
4  2021          4          10476933    12861  10489794   Oct-Dic 2021

quito el periodo, ya que con los trimestres y los años, es un dato redundante

In [18]:
na_accesos_b_ancha_angosta_fijo=na_accesos_b_ancha_angosta_fijo.drop(['Periodo'],axis=1)

In [19]:
na_accesos_b_ancha_angosta_fijo.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Año               36 non-null     int64
 1   Trimestre         36 non-null     int64
 2   Banda ancha fija  36 non-null     int64
 3   Dial up           36 non-null     int64
 4   Total             36 non-null     int64
dtypes: int64(5)
memory usage: 1.5 KB


In [20]:
na_accesos_b_ancha_angosta_fijo.to_csv("./csv_corregidos/na_accesos_b_ancha_angosta_fijo.csv", index=False)

### Accesos a banda ancha y banda angosta por provincia
Número de accesos al servicio de Internet fijo por banda ancha y banda angosta en cada provincia (trimestral)


In [21]:
prov_accesos_b_ancha_angosta_fijo.head()

Año  Trimestre        Provincia  Banda ancha fija  Dial up    Total
0  2022          4     Buenos Aires           4788466   6199.0  4794665
1  2022          4  Capital Federal           1549781   2145.0  1551926
2  2022          4        Catamarca             70908      1.0    70909
3  2022          4            Chaco            143658      5.0   143663
4  2022          4           Chubut            167855    904.0   168759

In [22]:
prov_accesos_b_ancha_angosta_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               864 non-null    int64  
 1   Trimestre         864 non-null    int64  
 2   Provincia         864 non-null    object 
 3   Banda ancha fija  864 non-null    int64  
 4   Dial up           862 non-null    float64
 5   Total             864 non-null    int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 40.6+ KB


In [23]:
prov_accesos_b_ancha_angosta_fijo.to_csv("./csv_corregidos/prov_accesos_b_ancha_angosta_fijo.csv", index=False)

### Serie trimestral de accesos a Internet fijo por tecnología
Número de accesos al servicio de Internet fijo por tipo de tecnología. Total nacional (trimestral)


In [24]:
na_accesos_tipo_tecnologia_fijo.head()


Año  Trimestre     ADSL  Cablemodem  Fibra óptica  Wireless   Otros  \
0  2022          4  1315373     6037457       3060575    572969  221740   
1  2022          3  1395277     6031970       2871541    557110  235230   
2  2022          2  1468333     5979214       2723285    556243  231609   
3  2022          1  1533240     6073426       2219533    545814  251996   
4  2021          4  1657615     5984240       2072236    523107  252596   

      Total        Periodo  
0  11208114   Oct-Dic 2022  
1  11091128  Jul-Sept 2022  
2  10958684   Abr-Jun 2022  
3  10624009   Ene-Mar 2022  
4  10489794   Oct-Dic 2021

quito el periodo, ya que con los trimestres y los años, es un dato redundante

In [25]:
na_accesos_tipo_tecnologia_fijo=na_accesos_tipo_tecnologia_fijo.drop(['Periodo'],axis=1)

In [26]:
na_accesos_tipo_tecnologia_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   Año           36 non-null     int64
 1   Trimestre     36 non-null     int64
 2   ADSL          36 non-null     int64
 3   Cablemodem    36 non-null     int64
 4   Fibra óptica  36 non-null     int64
 5   Wireless      36 non-null     int64
 6   Otros         36 non-null     int64
 7   Total         36 non-null     int64
dtypes: int64(8)
memory usage: 2.4 KB


In [27]:
na_accesos_tipo_tecnologia_fijo.to_csv("./csv_corregidos/na_accesos_tipo_tecnologia_fijo.csv", index=False)

### Acceso a Internet fijo por tecnología y provincia
Número de accesos al servicio de Internet fijo por tipo de tecnología en cada provincia (trimestral)


In [28]:
prov_accesos_tipo_tecnologia_fijo.head()

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2022         4     Buenos Aires  325908.0   2767247.0     1499149.0   
1  2022         4  Capital Federal  138753.0   1238196.0      138548.0   
2  2022         4        Catamarca    8952.0     10998.0       47337.0   
3  2022         4            Chaco   24582.0     61265.0       47285.0   
4  2022         4           Chubut   45450.0     73923.0        9815.0   

   Wireless    Otros      Total  Unnamed: 9  Unnamed: 10      Unnamed: 11  \
0  129910.0  72451.0  4794665.0         NaN          NaN     Buenos Aires   
1    4603.0  31826.0  1551926.0         NaN          NaN  Capital Federal   
2    1379.0   2243.0    70909.0         NaN          NaN        Catamarca   
3    7957.0   2574.0   143663.0         NaN          NaN            Chaco   
4   30721.0   8850.0   168759.0         NaN          NaN           Chubut   

   Unnamed: 12  Unnamed: 13  Unnamed: 14  Unnamed: 15  Unnamed: 16  \
0     325908.0    2767247.0    1499149.0     129910.0      72451.0   
1     138753.0    1238196.0     138548.0       4603.0      31826.0   
2       8952.0      10998.0      47337.0       1379.0       2243.0   
3      24582.0      61265.0      47285.0       7957.0       2574.0   
4      45450.0      73923.0       9815.0      30721.0       8850.0   

   Unnamed: 17  
0    4794665.0  
1    1551926.0  
2      70909.0  
3     143663.0  
4     168759.0

In [29]:
prov_accesos_tipo_tecnologia_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 866 entries, 0 to 865
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           865 non-null    object 
 1   Trimestre     865 non-null    object 
 2   Provincia     864 non-null    object 
 3   ADSL          864 non-null    float64
 4   Cablemodem    864 non-null    float64
 5   Fibra óptica  864 non-null    float64
 6   Wireless      864 non-null    float64
 7   Otros         864 non-null    float64
 8   Total         864 non-null    float64
 9   Unnamed: 9    0 non-null      float64
 10  Unnamed: 10   0 non-null      float64
 11  Unnamed: 11   25 non-null     object 
 12  Unnamed: 12   25 non-null     float64
 13  Unnamed: 13   25 non-null     float64
 14  Unnamed: 14   25 non-null     float64
 15  Unnamed: 15   25 non-null     float64
 16  Unnamed: 16   25 non-null     float64
 17  Unnamed: 17   25 non-null     float64
dtypes: float64(14), object(4)
memo

quitamos las columnas extra

In [30]:
to_drop=['Unnamed: 9','Unnamed: 10','Unnamed: 11','Unnamed: 12','Unnamed: 13','Unnamed: 14','Unnamed: 15','Unnamed: 16','Unnamed: 17']
prov_accesos_tipo_tecnologia_fijo=prov_accesos_tipo_tecnologia_fijo.drop(to_drop,axis=1)


In [31]:
prov_accesos_tipo_tecnologia_fijo.head()


Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2022         4     Buenos Aires  325908.0   2767247.0     1499149.0   
1  2022         4  Capital Federal  138753.0   1238196.0      138548.0   
2  2022         4        Catamarca    8952.0     10998.0       47337.0   
3  2022         4            Chaco   24582.0     61265.0       47285.0   
4  2022         4           Chubut   45450.0     73923.0        9815.0   

   Wireless    Otros      Total  
0  129910.0  72451.0  4794665.0  
1    4603.0  31826.0  1551926.0  
2    1379.0   2243.0    70909.0  
3    7957.0   2574.0   143663.0  
4   30721.0   8850.0   168759.0

In [32]:
prov_accesos_tipo_tecnologia_fijo.to_csv("./csv_corregidos/prov_accesos_tipo_tecnologia_fijo.csv", index=False)

### Velocidad Media de bajada de Internet fijo
Serie histórica de la velocidad media de descarga de Internet nacional (trimestral)


In [33]:
na_velocidad_media_bajada.head()


Año  Trimestre  Mbps (Media de bajada)    Trimestre.1
0  2022          4                  106.56   Oct-Dic 2022
1  2022          3                   62.46  Jul-Sept 2022
2  2022          2                   58.44   Abr-Jun 2022
3  2022          1                   55.11   Ene-Mar 2022
4  2021          4                   52.34   Oct-Dic 2021

quito el Trimestre.1, ya que con los trimestres y los años, es un dato redundante

In [34]:
na_velocidad_media_bajada=na_velocidad_media_bajada.drop(['Trimestre.1'],axis=1)

In [35]:
na_velocidad_media_bajada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 3 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     36 non-null     int64  
 1   Trimestre               36 non-null     int64  
 2   Mbps (Media de bajada)  36 non-null     float64
dtypes: float64(1), int64(2)
memory usage: 996.0 bytes


In [36]:
na_velocidad_media_bajada.to_csv("./csv_corregidos/na_velocidad_media_bajada.csv", index=False)

### Velocidad media de bajada de Internet fijo por provincia
Serie histórica de la velocidad media de descarga de Internet por provincia (trimestral)


In [37]:
prov_velocidad_media_bajada.head()


Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2022          4     Buenos Aires                  111.00
1  2022          4  Capital Federal                  182.86
2  2022          4        Catamarca                   83.33
3  2022          4            Chaco                  104.02
4  2022          4           Chubut                   15.92

In [38]:
prov_velocidad_media_bajada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     864 non-null    int64  
 1   Trimestre               864 non-null    int64  
 2   Provincia               864 non-null    object 
 3   Mbps (Media de bajada)  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


In [39]:
prov_velocidad_media_bajada.to_csv("./csv_corregidos/prov_velocidad_media_bajada.csv", index=False)

### Distribución de los accesos totales nacionales a Internet fijo por velocidad
Distribución de los accesos totales nacionales a Internet fijo por velocidad de bajada (último trimestre disponible)

In [40]:
na_dis_accesos_rango_vel_bajada.head()

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0  2022          4           31981                    85668   
1  2022          3           33013                    96727   
2  2022          2           33667                    99498   
3  2022          1           34890                   104840   
4  2021          4           41262                    28521   

   Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
0                1075272                 1109421                   765015   
1                1161370                 1151906                   823505   
2                1193090                 1197030                   856562   
3                1263273                 1209148                   967508   
4                1413208                 1245333                   976539   

   Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS     Total  
0                   335573         7410902  394282  11208114  
1                   479822         6995750  349035  11091128  
2                   485321         6741922  351594  10958684  
3                   509830         6336187  198333  10624009  
4                   558358         6032322  194251  10489794

In [41]:
na_dis_accesos_rango_vel_bajada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   Año                      36 non-null     int64
 1   Trimestre                36 non-null     int64
 2   Hasta 512 kbps           36 non-null     int64
 3   Entre 512 Kbps y 1 Mbps  36 non-null     int64
 4   Entre 1 Mbps y 6 Mbps    36 non-null     int64
 5   Entre 6 Mbps y 10 Mbps   36 non-null     int64
 6   Entre 10 Mbps y 20 Mbps  36 non-null     int64
 7   Entre 20 Mbps y 30 Mbps  36 non-null     int64
 8   Más de 30 Mbps           36 non-null     int64
 9   OTROS                    36 non-null     int64
 10  Total                    36 non-null     int64
dtypes: int64(11)
memory usage: 3.2 KB


In [42]:
na_dis_accesos_rango_vel_bajada.to_csv("./csv_corregidos/na_dis_accesos_rango_vel_bajada.csv", index=False)

### Acceso a Internet Fijo por rangos de velocidad de bajada y provincia
Número de accesos al servicio de Internet fijo por velocidad de bajada en cada provincia (trimestral)


In [43]:
prov_dis_accesos_rango_vel_bajada.head()


Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2022          4     Buenos Aires         28934.0              23347.0   
1  2022          4  Capital Federal           516.0               5096.0   
2  2022          4        Catamarca            71.0                315.0   
3  2022          4            Chaco           461.0                771.0   
4  2022          4           Chubut           109.0               1614.0   

   + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0           259866.0            281550.0             244555.0   
1            31320.0             64653.0              36336.0   
2             2716.0              5028.0               4638.0   
3            16067.0             17846.0               6704.0   
4            46457.0             31007.0              34728.0   

   + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
0              98670.0  3760109.0   97634.0  4794665.0  
1               8116.0  1299117.0  106772.0  1551926.0  
2               3122.0    52739.0    2280.0    70909.0  
3               3039.0    94866.0    3909.0   143663.0  
4              15770.0    19048.0   20026.0   168759.0

In [44]:
nuevos_nombres = [
    'Año', 'Trimestre', 'Provincia',
    'Hasta 512 kbps', 'Entre 512 Kbps y 1 Mbps', 'Entre 1 Mbps y 6 Mbps',
    'Entre 6 Mbps y 10 Mbps', 'Entre 10 Mbps y 20 Mbps', 'Entre 20 Mbps y 30 Mbps',
    'Más de 30 Mbps', 'OTROS', 'Total'
]

# Asignar los nuevos nombres a las columnas del DataFrame
prov_dis_accesos_rango_vel_bajada.columns = nuevos_nombres

In [45]:
prov_dis_accesos_rango_vel_bajada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Año                      864 non-null    int64  
 1   Trimestre                864 non-null    int64  
 2   Provincia                864 non-null    object 
 3   Hasta 512 kbps           864 non-null    float64
 4   Entre 512 Kbps y 1 Mbps  864 non-null    float64
 5   Entre 1 Mbps y 6 Mbps    864 non-null    float64
 6   Entre 6 Mbps y 10 Mbps   864 non-null    float64
 7   Entre 10 Mbps y 20 Mbps  864 non-null    float64
 8   Entre 20 Mbps y 30 Mbps  864 non-null    float64
 9   Más de 30 Mbps           864 non-null    float64
 10  OTROS                    858 non-null    float64
 11  Total                    864 non-null    float64
dtypes: float64(9), int64(2), object(1)
memory usage: 81.1+ KB


In [46]:
prov_dis_accesos_rango_vel_bajada.to_csv("./csv_corregidos/prov_dis_accesos_rango_vel_bajada.csv", index=False)

### Accesos a Internet fijo por velocidad bajada y provincia
Número de accesos al servicio de Internet fijo por velocidad de bajada en cada provincia


In [47]:
prov_dis_accesos_todo_vel_bajada.head()


Año  Trimestre        Provincia     Otros  0,256 Mbps  0,375 Mbps  \
0  2022.0        4.0     Buenos Aires   97634.0        18.0         0.0   
1  2022.0        4.0  Capital Federal  106772.0        31.0         0.0   
2  2022.0        4.0        Catamarca    2280.0         0.0         0.0   
3  2022.0        4.0            Chaco    3909.0         0.0       225.0   
4  2022.0        4.0           Chubut   20026.0         8.0         0.0   

   0,625 Mbps  0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  2048 Mbps  2529 Mbps  \
0         0.0   28724.0       192.0     4611.0  ...        0.0        0.0   
1         0.0     451.0        34.0     1307.0  ...        0.0        0.0   
2         0.0      34.0        37.0        0.0  ...        0.0        0.0   
3       111.0     234.0         2.0        0.0  ...        0.0        0.0   
4         0.0      69.0        32.0      836.0  ...        0.0        0.0   

   3072 Mbps  4000 Mbps  5000 Mbps  5125 Mbps  6000 Mbps  7000 Mbps  \
0        0.0        0.0        0.0        0.0        0.0        0.0   
1        0.0        0.0        0.0        0.0        0.0        0.0   
2        0.0        0.0        0.0        0.0        0.0        0.0   
3        0.0        0.0        0.0        0.0        0.0        0.0   
4        0.0        0.0        0.0        0.0        0.0        0.0   

   8000 Mbps  10000 Mbps  
0        0.0         0.0  
1        0.0         0.0  
2        0.0         0.0  
3        0.0         0.0  
4        0.0         0.0  

[5 rows x 186 columns]

In [48]:
prov_dis_accesos_todo_vel_bajada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 505 entries, 0 to 504
Columns: 186 entries, Año to 10000 Mbps
dtypes: float64(184), int64(1), object(1)
memory usage: 734.0+ KB


In [49]:
prov_dis_accesos_todo_vel_bajada.to_csv("./csv_corregidos/prov_dis_accesos_todo_vel_bajada.csv", index=False)

### Ingresos trimestrales por la prestación del servicio de Internet fijo
Ingresos trimestrales de los operadores por el servicio de Internet fijo

In [50]:
na_ingresos_operadores_fijo.head()

Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2022          4               7.244348e+07   Oct-Dic 2022
1  2022          3               6.408678e+07  Jul-Sept 2022
2  2022          2               6.041950e+07   Abr-Jun 2022
3  2022          1               5.521931e+07   Ene-Mar 2022
4  2021          4               4.546789e+07   Oct-Dic 2021

quito el Periodo, ya que con los trimestres y los años, es un dato redundante

In [51]:
na_ingresos_operadores_fijo=na_ingresos_operadores_fijo.drop(['Periodo'],axis=1)

In [52]:
na_ingresos_operadores_fijo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 3 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        36 non-null     int64  
 1   Trimestre                  36 non-null     int64  
 2   Ingresos (miles de pesos)  36 non-null     float64
dtypes: float64(1), int64(2)
memory usage: 996.0 bytes


In [53]:
na_ingresos_operadores_fijo.to_csv("./csv_corregidos/na_ingresos_operadores_fijo.csv", index=False)

### Accesos a Internet fijo por velocidad de bajada y localidad
Número de accesos al servicio de Internet fijo por velocidad de bajada en cada localidad declarada


In [54]:
loc_dis_accesos_todo_vel_bajada_sin_t.head()


Provincia     Partido          Localidad Link Indec  Otros  0,256 Mbps  \
0  BUENOS AIRES  25 de Mayo         25 de Mayo    6854100    NaN         NaN   
1  BUENOS AIRES  25 de Mayo    Agustín Mosconi    6854010    NaN         NaN   
2  BUENOS AIRES  25 de Mayo          Del Valle    6854020    NaN         NaN   
3  BUENOS AIRES  25 de Mayo          Ernestina    6854030    NaN         NaN   
4  BUENOS AIRES  25 de Mayo  Gobernador Ugarte    6854040    NaN         NaN   

   0,375 Mbps  0,5 Mbps  0,512 Mbps  0,625 Mbps  ...  600 Mbps  680 Mbps  \
0         NaN       2.0         NaN         NaN  ...       NaN       NaN   
1         NaN       NaN         NaN         NaN  ...       NaN       NaN   
2         NaN       1.0         NaN         NaN  ...       NaN       NaN   
3         NaN       NaN         NaN         NaN  ...       NaN       NaN   
4         NaN       NaN         NaN         NaN  ...       NaN       NaN   

   700 Mbps  800 Mbps  850 Mbps  900 Mbps  950 Mbps  1000 Mbps  1024 Mbps  \
0       NaN       NaN       NaN       NaN       NaN        NaN        NaN   
1       NaN       NaN       NaN       NaN       NaN        NaN        NaN   
2       NaN       NaN       NaN       NaN       NaN        NaN        NaN   
3       NaN       NaN       NaN       NaN       NaN        NaN        NaN   
4       NaN       NaN       NaN       NaN       NaN        NaN        NaN   

   Total general  
0         6208.0  
1           59.0  
2          192.0  
3           66.0  
4          181.0  

[5 rows x 134 columns]

In [55]:
loc_dis_accesos_todo_vel_bajada_sin_t.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Columns: 134 entries, Provincia to Total general
dtypes: float64(130), object(4)
memory usage: 3.2+ MB


In [56]:
loc_dis_accesos_todo_vel_bajada_sin_t.to_csv("./csv_corregidos/loc_dis_accesos_todo_vel_bajada_sin_t.csv", index=False)

### Accesos a Internet fijo por tecnología y localidad
Número de accesos al servicio de Internet fijo por tecnología en cada localidad declarada

In [57]:
loc_accesos_tipo_tecnologia_fijo_sin_t.head()

Provincia     Partido               Localidad Link Indec  ADSL  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo    6854100   874   
1  BUENOS AIRES  25 de Mayo               Del Valle    6854020   181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte    6854040     0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra    6854060     0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde    6854050     0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  \
0        4542        0             3      0        852      0       706   
1           0        0             0     10          1      0         0   
2           0        0             0      0          0      0       181   
3         782        0             6    167          0      0       327   
4           0        0             0      0          0      0         6   

   Total general  
0           6977  
1            192  
2            181  
3           1282  
4              6

In [58]:
loc_accesos_tipo_tecnologia_fijo_sin_t.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3137 entries, 0 to 3136
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Provincia      3137 non-null   object
 1   Partido        3137 non-null   object
 2   Localidad      3137 non-null   object
 3   Link Indec     3137 non-null   object
 4   ADSL           3137 non-null   int64 
 5   CABLEMODEM     3137 non-null   int64 
 6   DIAL UP        3137 non-null   int64 
 7   FIBRA OPTICA   3137 non-null   int64 
 8   OTROS          3137 non-null   int64 
 9   SATELITAL      3137 non-null   int64 
 10  WIMAX          3137 non-null   int64 
 11  WIRELESS       3137 non-null   int64 
 12  Total general  3137 non-null   int64 
dtypes: int64(9), object(4)
memory usage: 318.7+ KB


In [59]:
loc_accesos_tipo_tecnologia_fijo_sin_t.to_csv("./csv_corregidos/loc_accesos_tipo_tecnologia_fijo_sin_t.csv", index=False)

### Listado de localidades con conectividad a internet.
Listado de localidades con conectividad a internet, con detalle por tipo de conexión.


In [60]:
loc_listado_conectividad_conexion_sin_t.head()

Provincia     Partido                                Localidad  \
0  BUENOS AIRES  9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   
1  BUENOS AIRES  9 de Julio                        Carlos María Naón   
2  BUENOS AIRES  9 de Julio                                 Dudignac   
3  BUENOS AIRES  9 de Julio                 La Aurora (Est. La Niña)   
4  BUENOS AIRES  9 de Julio           Manuel B. Gonnet (Est. French)   

   Población ADSL Cablemódem Dial Up Fibra óptica Satelital Wireless  \
0       1853   SI         --      --           --        --       --   
1        497   --         --      --           --        --       --   
2       2670   --         --      SI           --        --       SI   
3        531   --         --      --           --        --       --   
4        748   SI         --      --           SI        --       --   

  Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0             SI  SI  SI  6588010.0 -35.292982 -61.407254  
1             SI  SI  SI  6588020.0 -35.239499 -60.825136  
2             SI  SI  SI  6588040.0 -35.651193 -60.709896  
3             --  --  SI  6588050.0 -35.407947 -61.210567  
4             SI  SI  SI  6588060.0 -35.520141 -60.998949

In [61]:
loc_listado_conectividad_conexion_sin_t.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4311 entries, 0 to 4310
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4311 non-null   object 
 1   Partido         4311 non-null   object 
 2   Localidad       4311 non-null   object 
 3   Población       4311 non-null   int64  
 4   ADSL            4311 non-null   object 
 5   Cablemódem      4311 non-null   object 
 6   Dial Up         4311 non-null   object 
 7   Fibra óptica    4311 non-null   object 
 8   Satelital       4311 non-null   object 
 9   Wireless        4311 non-null   object 
 10  Telefonía Fija  4311 non-null   object 
 11  3G              4311 non-null   object 
 12  4G              4311 non-null   object 
 13  Link            4299 non-null   float64
 14  Latitud         4311 non-null   float64
 15  Longitud        4311 non-null   float64
dtypes: float64(3), int64(1), object(12)
memory usage: 539.0+ KB


modifico las columnas para que tengan valores binarios y sean mas faciles de procesar

In [62]:
# Columnas que contienen "SI" o "--" que deseas convertir a valores binarios.
columnas_a_procesar = ['ADSL', 'Cablemódem', 'Dial Up', 'Fibra óptica', 'Satelital', 'Wireless', 'Telefonía Fija', '3G', '4G']

# Diccionario de mapeo para convertir "SI" y "--" a valores binarios (1 o 0).
mapeo_valores = {"SI": 1, "--": 0}

# Aplicar el mapeo a las columnas relevantes
loc_listado_conectividad_conexion_sin_t[columnas_a_procesar] = loc_listado_conectividad_conexion_sin_t[columnas_a_procesar].applymap(lambda x: mapeo_valores.get(x, x))

In [63]:
loc_listado_conectividad_conexion_sin_t.head()

Provincia     Partido                                Localidad  \
0  BUENOS AIRES  9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   
1  BUENOS AIRES  9 de Julio                        Carlos María Naón   
2  BUENOS AIRES  9 de Julio                                 Dudignac   
3  BUENOS AIRES  9 de Julio                 La Aurora (Est. La Niña)   
4  BUENOS AIRES  9 de Julio           Manuel B. Gonnet (Est. French)   

   Población  ADSL  Cablemódem  Dial Up  Fibra óptica  Satelital  Wireless  \
0       1853     1           0        0             0          0         0   
1        497     0           0        0             0          0         0   
2       2670     0           0        1             0          0         1   
3        531     0           0        0             0          0         0   
4        748     1           0        0             1          0         0   

   Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0               1   1   1  6588010.0 -35.292982 -61.407254  
1               1   1   1  6588020.0 -35.239499 -60.825136  
2               1   1   1  6588040.0 -35.651193 -60.709896  
3               0   0   1  6588050.0 -35.407947 -61.210567  
4               1   1   1  6588060.0 -35.520141 -60.998949

In [64]:
loc_listado_conectividad_conexion_sin_t.to_csv("./csv_corregidos/loc_listado_conectividad_conexion_sin_t.csv", index=False)

# Selección de los datos y comunicacion entre los mismos.

#### Carga de datasets modificados
Separamos los datos es nacion(na), provincia(prov) y localidad(loc)

In [65]:
#nacional
na_accesos_cada_100_hogares_fijo=pd.read_csv('./csv_corregidos/na_accesos_cada_100_hogares_fijo.csv')
na_accesos_b_ancha_angosta_fijo= pd.read_csv('./csv_corregidos/na_accesos_b_ancha_angosta_fijo.csv')
na_accesos_tipo_tecnologia_fijo= pd.read_csv('./csv_corregidos/na_accesos_tipo_tecnologia_fijo.csv')
na_velocidad_media_bajada= pd.read_csv('./csv_corregidos/na_velocidad_media_bajada.csv')
na_dis_accesos_rango_vel_bajada= pd.read_csv('./csv_corregidos/na_dis_accesos_rango_vel_bajada.csv')
na_ingresos_operadores_fijo= pd.read_csv('./csv_corregidos/na_ingresos_operadores_fijo.csv')
#provincial
prov_accesos_cada_100_hogares_fijo=pd.read_csv('./csv_corregidos/prov_accesos_cada_100_hogares_fijo.csv')
prov_accesos_b_ancha_angosta_fijo = pd.read_csv('./csv_corregidos/prov_accesos_b_ancha_angosta_fijo.csv')
prov_accesos_tipo_tecnologia_fijo = pd.read_csv('./csv_corregidos/prov_accesos_tipo_tecnologia_fijo.csv')
prov_velocidad_media_bajada = pd.read_csv('./csv_corregidos/prov_velocidad_media_bajada.csv')
prov_dis_accesos_rango_vel_bajada = pd.read_csv('./csv_corregidos/prov_dis_accesos_rango_vel_bajada.csv')
prov_dis_accesos_todo_vel_bajada = pd.read_csv('./csv_corregidos/prov_dis_accesos_todo_vel_bajada.csv')
#localidad
loc_dis_accesos_todo_vel_bajada_sin_t = pd.read_csv('./csv_corregidos/loc_dis_accesos_todo_vel_bajada_sin_t.csv')
loc_accesos_tipo_tecnologia_fijo_sin_t = pd.read_csv('./csv_corregidos/loc_accesos_tipo_tecnologia_fijo_sin_t.csv')
loc_listado_conectividad_conexion_sin_t = pd.read_csv('./csv_corregidos/loc_listado_conectividad_conexion_sin_t.csv')

#### Nacional
-_Los df a nivel nacional cuentan de 36 observaciones. Sin valores faltantes observables._

-_Se puede ver que los datos estan agrupados por en 4 trimestres por año y van desde el 2014 al 2018._

Vemos que todas las tablas nacionales se dividen en 4 trimestres por año y van desde el 2014 al 2018

In [66]:
print(na_accesos_cada_100_hogares_fijo.iloc[:, :2].describe())
print(na_accesos_b_ancha_angosta_fijo.iloc[:, :2].describe())
print(na_accesos_tipo_tecnologia_fijo.iloc[:, :2].describe())

               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000
               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000
               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000


In [67]:
print(na_velocidad_media_bajada.iloc[:, :2].describe())
print(na_dis_accesos_rango_vel_bajada.iloc[:, :2].describe())
print(na_ingresos_operadores_fijo.iloc[:, :2].describe())

               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000
               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000
               Año  Trimestre
count    36.000000  36.000000
mean   2018.000000   2.500000
std       2.618615   1.133893
min    2014.000000   1.000000
25%    2016.000000   1.750000
50%    2018.000000   2.500000
75%    2020.000000   3.250000
max    2022.000000   4.000000


#### Provincial
-_Los df a nivel provincial cuentan de 864 observaciones. exceptuando 'prov_dis_accesos_todo_vel_bajada' que cuenta con 504 observaciones_

-_'prov_dis_accesos_todo_vel_bajada' no se va a utilizar, ya que 'prov_dis_accesos_rango_vel_bajada' contiene datos similares y abarca todos los años_

-_Se puede ver que los datos estan agrupados por en 4 trimestres por año y van desde el 2014 al 2018._

cantidad de datos

In [68]:
#provincial
print(prov_accesos_cada_100_hogares_fijo.shape[0])
print(prov_accesos_b_ancha_angosta_fijo.shape[0])
print(prov_accesos_tipo_tecnologia_fijo.shape[0])
print(prov_velocidad_media_bajada.shape[0])
print(prov_dis_accesos_rango_vel_bajada.shape[0])
print(prov_dis_accesos_todo_vel_bajada.shape[0])

864
864
866
864
864
505


Vemos que al igual que las tablas a nivel nacional, todas las tablas nacionales se dividen en 4 trimestres por año y van desde el 2014 al 2018

In [69]:
print(prov_accesos_cada_100_hogares_fijo.iloc[:, :2].describe())
print(prov_accesos_b_ancha_angosta_fijo.iloc[:, :2].describe())
print(prov_accesos_tipo_tecnologia_fijo.iloc[:, :2].describe())


               Año   Trimestre
count   864.000000  864.000000
mean   2018.000000    2.500000
std       2.583484    1.118682
min    2014.000000    1.000000
25%    2016.000000    1.750000
50%    2018.000000    2.500000
75%    2020.000000    3.250000
max    2022.000000    4.000000
               Año   Trimestre
count   864.000000  864.000000
mean   2018.000000    2.500000
std       2.583484    1.118682
min    2014.000000    1.000000
25%    2016.000000    1.750000
50%    2018.000000    2.500000
75%    2020.000000    3.250000
max    2022.000000    4.000000
         Año Trimestre
count    865       865
unique    11         8
top     2022         4
freq      96       216


In [70]:
print(prov_velocidad_media_bajada.iloc[:, :2].describe())
print(prov_dis_accesos_rango_vel_bajada.iloc[:, :2].describe())
print(prov_dis_accesos_todo_vel_bajada.iloc[:, :2].describe())

               Año   Trimestre
count   864.000000  864.000000
mean   2018.000000    2.500000
std       2.583484    1.118682
min    2014.000000    1.000000
25%    2016.000000    1.750000
50%    2018.000000    2.500000
75%    2020.000000    3.250000
max    2022.000000    4.000000
               Año   Trimestre
count   864.000000  864.000000
mean   2018.000000    2.500000
std       2.583484    1.118682
min    2014.000000    1.000000
25%    2016.000000    1.750000
50%    2018.000000    2.500000
75%    2020.000000    3.250000
max    2022.000000    4.000000
               Año   Trimestre
count   504.000000  504.000000
mean   2019.857143    2.571429
std       1.522341    1.138019
min    2017.000000    1.000000
25%    2019.000000    2.000000
50%    2020.000000    3.000000
75%    2021.000000    4.000000
max    2022.000000    4.000000


Vemos que el df 'prov_dis_accesos_rango_vel_bajada' y el df 'prov_dis_accesos_todo_vel_bajada' contienen datos similares, pero el primero contiene los datos en rangos definidos y tiene los datos desde 2014, mientras que el segundo solo tiene datos desde el año 2017, por lo que no se va a utilizar

In [71]:
prov_dis_accesos_rango_vel_bajada.describe()

Año   Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
count   864.000000  864.000000      864.000000               864.000000   
mean   2018.000000    2.500000     2065.484954              9960.253472   
std       2.583484    1.118682    13459.419708             21992.818620   
min    2014.000000    1.000000        6.000000                 0.000000   
25%    2016.000000    1.750000       39.000000               312.000000   
50%    2018.000000    2.500000      109.000000              2536.000000   
75%    2020.000000    3.250000      541.250000              8056.250000   
max    2022.000000    4.000000   238920.000000            171244.480000   

       Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
count           8.640000e+02              864.000000               864.000000   
mean            1.480209e+05            37641.817130             38114.030093   
std             3.438435e+05            65383.130836             93024.185664   
min             2.531000e+03                0.000000                 0.000000   
25%             2.744150e+04             2929.000000              1579.750000   
50%             4.783750e+04            11247.000000              9067.000000   
75%             8.639200e+04            38753.250000             26408.000000   
max             2.299705e+06           403575.000000            886678.000000   

       Entre 20 Mbps y 30 Mbps  Más de 30 Mbps         OTROS         Total  
count               864.000000    8.640000e+02     858.00000  8.640000e+02  
mean              20793.402778    8.555756e+04    5289.20979  3.474059e+05  
std               73859.534388    3.635417e+05   13011.53175  7.450600e+05  
min                   0.000000    0.000000e+00   -1945.00000  1.240600e+04  
25%                  25.750000    8.000000e+00       0.00000  5.274050e+04  
50%                1146.000000    1.019000e+03       0.00000  1.052355e+05  
75%                9567.500000    2.247475e+04    5146.00000  1.791482e+05  
max              949093.000000    3.760109e+06  120464.00000  4.794665e+06

In [72]:
prov_dis_accesos_todo_vel_bajada.describe()

Año   Trimestre          Otros  0,256 Mbps  0,375 Mbps  \
count   504.000000  504.000000     504.000000  504.000000  504.000000   
mean   2019.857143    2.571429    9132.545635    3.240079    5.357143   
std       1.522341    1.138019   16245.555828   11.108175   34.336532   
min    2017.000000    1.000000       0.000000    0.000000    0.000000   
25%    2019.000000    2.000000     537.750000    0.000000    0.000000   
50%    2020.000000    3.000000    3987.000000    0.000000    0.000000   
75%    2021.000000    4.000000   11535.250000    0.250000    0.000000   
max    2022.000000    4.000000  120987.000000  102.000000  225.000000   

       0,625 Mbps       0,5 Mbps  0,512 Mbps     0,75 Mbps        1 Mbps  ...  \
count  504.000000     504.000000  504.000000    504.000000    504.000000  ...   
mean     2.642857    2491.728175   39.031746   1312.303571   5230.615079  ...   
std     16.939356   17503.933621   74.020006   5171.437899   7845.964066  ...   
min      0.000000       6.000000    0.000000      0.000000      0.000000  ...   
25%      0.000000      20.750000    0.000000      0.000000    902.250000  ...   
50%      0.000000      47.000000   10.500000      1.000000   2364.000000  ...   
75%      0.000000      82.250000   57.000000    313.000000   5760.250000  ...   
max    111.000000  238902.000000  609.000000  35185.000000  49473.000000  ...   

        2048 Mbps   2529 Mbps   3072 Mbps   4000 Mbps   5000 Mbps   5125 Mbps  \
count  504.000000  504.000000  504.000000  504.000000  504.000000  504.000000   
mean     8.547619    0.039683    2.251984    0.134921    0.073413    0.007937   
std     85.315283    0.890871   22.519662    1.342696    0.800452    0.088821   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
max    860.000000   20.000000  227.000000   15.000000    9.000000    1.000000   

        6000 Mbps   7000 Mbps   8000 Mbps  10000 Mbps  
count  504.000000  504.000000  504.000000  504.000000  
mean     0.067460    0.196429    0.015873    0.031746  
std      0.672827    1.602405    0.177642    0.355284  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000    0.000000  
50%      0.000000    0.000000    0.000000    0.000000  
75%      0.000000    0.000000    0.000000    0.000000  
max      8.000000   18.000000    2.000000    4.000000  

[8 rows x 185 columns]

#### Localidad
-_No se van a usar los df que contienen localidades ya que no tienen fechas, horas o trimestres de los que se pueda guiar_
-_Tambien vemos que todas las tablas tienen diferencias en la cantidad de datos, por lo que habria discrepancia en las localidades vistas_

vemos que todas las tablas tienen diferencias en la cantidad de datos

In [73]:
print(loc_dis_accesos_todo_vel_bajada_sin_t.shape[0])
print(loc_accesos_tipo_tecnologia_fijo_sin_t.shape[0])
print(loc_listado_conectividad_conexion_sin_t.shape[0])

3095
3137
4311


Vemos que los df comparten la columnas 'Provincia', 'Partido' y 'Localidad'. pero no tienen una información temporal, como fechas o trimestres. La ausencia de datos temporales limita la capacidad de realizar análisis de tendencias a lo largo del tiempo y afecta la profundidad de los resultados.

In [74]:
nombres_columnas = loc_dis_accesos_todo_vel_bajada_sin_t.columns
print(nombres_columnas)
nombres_columnas = loc_dis_accesos_todo_vel_bajada_sin_t.columns
print(nombres_columnas)
nombres_columnas = loc_accesos_tipo_tecnologia_fijo_sin_t.columns
print(nombres_columnas)

Index(['Provincia', 'Partido', 'Localidad', 'Link Indec', 'Otros',
       '0,256 Mbps', '0,375 Mbps', '0,5 Mbps', '0,512 Mbps', '0,625 Mbps',
       ...
       '600 Mbps', '680 Mbps', '700 Mbps', '800 Mbps', '850 Mbps', '900 Mbps',
       '950 Mbps', '1000 Mbps', '1024 Mbps', 'Total general'],
      dtype='object', length=134)
Index(['Provincia', 'Partido', 'Localidad', 'Link Indec', 'Otros',
       '0,256 Mbps', '0,375 Mbps', '0,5 Mbps', '0,512 Mbps', '0,625 Mbps',
       ...
       '600 Mbps', '680 Mbps', '700 Mbps', '800 Mbps', '850 Mbps', '900 Mbps',
       '950 Mbps', '1000 Mbps', '1024 Mbps', 'Total general'],
      dtype='object', length=134)
Index(['Provincia', 'Partido', 'Localidad', 'Link Indec', 'ADSL', 'CABLEMODEM',
       'DIAL UP', 'FIBRA OPTICA', 'OTROS', 'SATELITAL', 'WIMAX', 'WIRELESS',
       'Total general'],
      dtype='object')
